cdn_lambda/functions/origin_response/origin_response.py

```diff
@@ -0,0 +1,63 @@
+import json
+import logging
+import re
+
+LOG = logging.getLogger("origin-response")
+
+
+class LambdaClient(object):
+    def __init__(self, conf_file="lambda_conf.json"):
+        self._conf_file = conf_file
+        self._conf = None
+
+    @property
+    def conf(self):
+        if not self._conf:
+            with open(self._conf_file, "r") as json_file:
+                self._conf = json.load(json_file)
+
+        return self._conf
+
+    def handler(self, event, context):
+        # pylint: disable=unused-argument
+
+        request = event["Records"][0]["cf"]["request"]
+        response = event["Records"][0]["cf"]["response"]
+
+        max_age = self.conf["headers"]["max_age"]
+        max_age_pattern_whitelist = [
+            ".+/PULP_MANIFEST",
+            ".+/listing",
+            ".+/repodata/repomd.xml",
+            ".+/ostree/repo/refs/heads/.*/.*",
+        ]
+
+        try:
+            original_uri = request["headers"]["exodus-original-uri"][0][
+                "value"
+            ]
+
+        except (KeyError):
+            LOG.debug(
+                "Could not add Cache-Control header 'exodus-original-uri'"
```

Comment:
This log message seems to be a bit mangled, it reads like 'exodus-original-uri' is a cache-control header, and that we failed to add that header, both of which are not accurate.

Also it should probably have exc_info so that the exception isn't lost entirely, because actually you don't know for certain that the KeyError was raised due to missing exodus-original-uri.

Would something like this be more appropriate?

```suggestion
                "Could not read exodus-original-uri from response", exc_info=1
```

exc_info=1 means it will log backtrace of the exception currently being handled.